### DISCUSSION TITLE
Twice requests 'initialize'

### DISCUSSION THREAD

**Discussion by onriv** [redhat-developer/lsp4ij#478] [Q&A] [answered]
Hi, I occasionally encounter an issue where the trace tool window shows “initializing” twice, as seen in the screenshot below:
![image](https://github.com/user-attachments/assets/fabae200-a8fc-4d79-aeb3-4b404b4211c5)
Could you help me understand why this happens? The exception in the screenshot is `java.io.IOException: The pipe is being closed`. Is this because the server has quit unexpectedly?

Additionally, why would there be a log entry like `[Trace - 23:58:33] Sending response <unknown> - (register_lean_watcher). Processing request took ?ms? Could this be due to a race condition`? I reviewed some code, such as LanguageServerWrapper and TracingMessageConsumer, but I still don’t have any clues. I’ve attached the full related log below:

[twice_initialize.log](https://github.com/user-attachments/files/16751499/twice_initialize.log)

**Comment by angelozerr** [comment, marked as answer]
> Could you help me understand why this happens? 

If there are 2 initializing, it means that server is started, stopped (because of some crash that you can see in stacktrace error) and server is restarted (because some LSP features are called after the crash and need the start of language server).

> Additionally, why would there be a log entry like [Trace - 23:58:33] Sending response <unknown> - (register_lean_watcher). Processing request took ?ms? 

We store in a Map the request by id and when we receive response we try to get the registered request by the id. If we cannot find the response with id, we display this kind of message.

I don't know why you have this usecase. It will require to debug your language server, but to be honnest with you I'm very busy for now.

**Reply by onriv** [reply]
Thanks for the detailed explanation! Yeah it will work after the retries. Not a problem maybe
